albumentations/core/bbox_utils.py

```diff
@@ -475,10 +475,7 @@ def filter_bboxes(
     filtered_bboxes = clipped_bboxes[mask]
 
     # If no bboxes pass the filter, return an empty array with the same number of columns as input
-    if len(filtered_bboxes) == 0:
-        return np.array([], dtype=np.float32)
-
-    return filtered_bboxes
+    return filtered_bboxes if len(filtered_bboxes) > 0 else np.array([], dtype=np.float32)
```

Comment:
Hey @ternaus , thanks for fixing the issue.
Regarding this line of code, notice that it creates the issue of returning arrays with different number of dimensions depending on the number of bounding boxes that remains.

- If there is at least one box in `filtered_boxes` the function will return an array of shape (n, 4), i.e. dim=2.
- If there is no box in `filtered_boxes` the function will return an array of shape (0,), i.e. dim=1.

**Ideally, if there are no boxes left, we should return an array of shape (0, 4).** Which ensures data shape consistency.
So far, whenever I use albumentations, I have to call `bboxes.reshape(-1, 4)`  to work around this issue. Otherwise I can't , for instance, concatenated two arrays of bounding boxes.

Regarding this line of code, can't we just return `filtered_boxes` straight away, i.e. remove this if clause? (Disclaimer: I haven't checked the shape of `filtered_boxes` when the number of boxes is zero)

Reply:
@gui-miotto 

Thanks. Fixed in https://github.com/albumentations-team/albumentations/pull/1949